bot/cogs/bot.py

```diff
@@ -150,11 +134,53 @@ def codeblock_stripping(self, msg: str):
 
                 # Strip again to remove the whitespace(s) left before the code
                 # If the msg looked like "Python <code>" before removing Python
-                # And strips REPL code out of the message if there is any
+                # Strips REPL code out of the message if there is any
+                # And tries to apply identation fixes to the code
                 content = self.repl_stripping(content.strip())
+                content = self.fix_overindented(content)
+
                 log.trace(f"Returning message.\n\n{content}\n\n")
                 return content
 
+    def fix_overindented(self, msg: str):
```

Comment:
hmmm. I'm not sure if this can be replaced altogether with it, but I have a feeling that [`textwrap.dedent`](https://docs.python.org/3/library/textwrap.html#textwrap.dedent) might prove very helpful here. see this:
```py
>>> print(dedent("""
...     if 2 + 2 == 5:
...             raise ArithmeticError("math broke")
...     else:
...             print("math is working")
... """))

if 2 + 2 == 5:
	raise ArithmeticError("math broke")
else:
	print("math is working")
```